src/java/org/apache/cassandra/cql3/restrictions/SimpleRestriction.java

```diff
@@ -217,7 +219,7 @@ public boolean isSupportedBy(Index index)
     @Override
     public List<ClusteringElements> values(QueryOptions options)
     {
-        assert operator == Operator.EQ || operator == Operator.IN || operator == Operator.ANN;
+        assert operator == Operator.EQ || operator == Operator.IN || operator == Operator.ANN || operator == Operator.NEQ || operator == Operator.NOT_IN;
```

Comment:
Nit: we can split the long line
```suggestion
        assert operator == Operator.EQ ||
               operator == Operator.NEQ ||
               operator == Operator.IN ||
               operator == Operator.NOT_IN ||
               operator == Operator.ANN;
```